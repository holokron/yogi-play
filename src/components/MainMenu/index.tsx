import React, { ReactElement } from "react";
import { Link } from "react-router-dom";
import { Nav, NavItem } from "reactstrap";

export interface Props {
  onLinkClick?(): void;
}

export default function MainMenu({
  onLinkClick = () => {},
}: Props): ReactElement<Props> {
  return (
    <Nav navbar className="mr-auto">
      <NavItem>
        <Link
          to="/ulubione"
          className="nav-link text-uppercase"
          onClick={onLinkClick}
        >
          Ulubione
        </Link>
      </NavItem>
      <NavItem>
        <a
          href="https://github.com/holokron/yogi-play/issues/new?labels=sounds&assignee=michalv8"
          rel="noopener noreferrer"
          target="_blank"
          className="nav-link text-uppercase"
        >
          Dodaj dźwięk
        </a>
      </NavItem>
    </Nav>
  );
}
