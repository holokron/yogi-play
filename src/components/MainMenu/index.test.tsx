import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import MainMenu from '.'

describe('@components/MainMenu', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <MainMenu />
            </MemoryRouter>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })
})