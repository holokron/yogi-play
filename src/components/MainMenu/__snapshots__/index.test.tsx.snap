// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/MainMenu renders correctly 1`] = `
<ul
  className="mr-auto navbar-nav"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link text-uppercase"
      href="/ulubione"
      onClick={[Function]}
    >
      Ulubione
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link text-uppercase"
      href="https://github.com/holokron/yogi-play/issues/new?labels=sounds&assignee=michalv8"
      rel="noopener noreferrer"
      target="_blank"
    >
      Dodaj dźwięk
    </a>
  </li>
</ul>
`;
